<launch>

  <!-- ====================== Detection Node ====================== -->
  <node name="detector_node" pkg="dopplertrack" type="detector_node" output="screen">
  
    <!-- Detection hyperparameters -->
    <param name="ground_height" value="-1.75" />
    <param name="default_bucket_size" value="0.8" />
    <param name="delta_rho" value="0.2" />
    <param name="delta_theta" value="0.4" />

    <!-- NMS hyperparameters -->
    <param name="iou_threshold" value="0.4" />
    <param name="coverage_threshold" value="0.5" />
    <param name="use_car_flag" value="true" />
    <param name="debug_flag" value="true" />

  </node>

  <!-- ====================== Tracking Node ====================== -->
  <node name="tracker_node" pkg="dopplertrack" type="tracker_node" output="screen">
    <param name="output_dir" value="$(find dopplertrack)/eval/output"/>
  </node>

  <!-- ====================== Publisher Node (Delayed Start) ====================== -->
  <node name="publisher_node"
        pkg="dopplertrack"
        type="publisher_node"
        output="screen"
        launch-prefix="bash -c 'sleep 5; exec $0 $@'">
    <param name="type" value="aq" />
    <param name="path" value="$(find dopplertrack)/data/original_bin/Aq-car" />
    <param name="start_frame" value="500" />
  </node>

</launch>